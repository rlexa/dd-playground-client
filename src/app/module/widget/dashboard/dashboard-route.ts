export enum DashboardRoute {
  Overview = 'overview',
  Graph = 'graph',
  Crypto = 'crypto',
  School = 'school',
  Game = 'game',
  Api = 'api',
  Playground = 'playground',
  Settings = 'settings',
}
