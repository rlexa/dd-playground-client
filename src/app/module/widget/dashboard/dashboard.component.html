<div class="content mat-app-background">
  <app-navigation-bar *ngIf="isVisibleHeader$ | async" [items]="navs$ | async"></app-navigation-bar>

  <main>
    <router-outlet></router-outlet>
  </main>

  <app-footer *ngIf="isVisibleFooter$ | async"></app-footer>
</div>

<!-- <div
  class="mat-app-background position-absolute match-parent"
  [ngStyle]="{
    display: 'grid',
    'grid-template-areas': GRID,
    'grid-template-columns': 'max-content auto',
    'grid-template-rows': 'max-content auto max-content'
  }"
>
  <div [ngStyle]="{'grid-area': GRID_HEAD, background: 'blue'}">
    <ng-container *ngIf="isVisibleHeader$ | async">
      <mat-toolbar color="primary">
        <span class="mat-toolbar-filler"></span>
        <nav class="mat-nav-list">
          <ng-container *ngFor="let ii of ROUTES">
            <a
              mat-button
              [style.min-width]="'auto'"
              [routerLink]="'/' + ROUTE_ROOT + '/' + ii.route"
              routerLinkActive="active"
              [matTooltip]="ii.label"
            >
              <i class="material-icons transition-transform active-scale">{{ ii.icon | icon }}</i>
            </a>
          </ng-container>
        </nav>
        <span class="mat-toolbar-filler"></span>
      </mat-toolbar>
    </ng-container>
  </div>

  <div [ngStyle]="{'grid-area': GRID_SIDE}" class="position-relative">
    <ng-container *ngIf="isVisibleSide$ | async">
      <mat-toolbar color="primary" class="toolbar-column">
        <nav *ngIf="(subRoute$ | async) && (subRoutes$ | async).length" class="mat-nav-list">
          <ng-container *ngFor="let ii of subRoutes$ | async">
            <a
              mat-button
              [style.minWidth]="'auto'"
              [routerLink]="'/' + ROUTE_ROOT + '/' + (subRoute$ | async) + '/' + ii.route"
              routerLinkActive="active"
              [matTooltip]="ii.label"
              matTooltipPosition="right"
            >
              <i class="material-icons transition-transform active-scale">{{ ii.icon | icon }}</i>
            </a>
          </ng-container>
        </nav>
        <span class="mat-toolbar-filler"></span>
      </mat-toolbar>
    </ng-container>
  </div>

  <main class="position-relative" [ngStyle]="{'grid-area': GRID_CONTENT}">
    <router-outlet></router-outlet>
  </main>

  <footer [ngStyle]="{'grid-area': GRID_FOOTER}">
    <div *ngIf="isVisibleFooter$ | async" ddFlexbox="hycr" class="border-top contenthv">
      <app-version class="margin-bottom-narrow"></app-version>
    </div>
  </footer>
</div> -->
