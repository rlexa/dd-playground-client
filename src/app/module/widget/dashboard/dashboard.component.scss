.content {
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;

  display: flex;
  flex-direction: column;

  > main {
    flex: 1 1 auto;
    position: relative;
  }
}
