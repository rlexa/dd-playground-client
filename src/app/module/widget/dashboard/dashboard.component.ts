import {ChangeDetectionStrategy, Component, OnDestroy} from '@angular/core';
import {DoneSubject, RxCleanup} from 'dd-rxjs';
import {map} from 'rxjs/operators';
import {
  ROUTE_BLOCKCHAIN,
  ROUTE_BUILDCONFIG,
  ROUTE_CONFIGURATION,
  ROUTE_CRYPTO,
  ROUTE_CURRENT,
  ROUTE_DASHBOARD,
  ROUTE_DEMO_GHIBLI,
  ROUTE_DEMO_MISC,
  ROUTE_DEMO_STATE,
  ROUTE_GAME,
  ROUTE_GAME_DOWN,
  ROUTE_GAME_SNAKE,
  ROUTE_GRAPH,
  ROUTE_OVERVIEW,
  ROUTE_PLAYGROUND,
  ROUTE_RENDER_CANVAS,
  ROUTE_SETTINGS,
  ROUTE_WALKER,
  ROUTE_GAME_MINESWEEPER,
} from 'src/app/routing';
import {RxStateService} from 'src/app/rx-state';

interface RouteDef {
  icon: string;
  label: string;
  route: string;
  subs: RouteDef[];
}

@Component({
  selector: 'app-dashboard',
  templateUrl: './dashboard.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class DashboardComponent implements OnDestroy {
  constructor(private rxState: RxStateService) {}

  @RxCleanup() private readonly done$ = new DoneSubject();

  readonly GRID_CONTENT = 'cont';
  readonly GRID_FOOTER = 'foot';
  readonly GRID_HEAD = 'head';
  readonly GRID_SIDE = 'side';
  readonly GRID = [
    [this.GRID_HEAD, this.GRID_HEAD],
    [this.GRID_SIDE, this.GRID_CONTENT],
    [this.GRID_FOOTER, this.GRID_FOOTER],
  ]
    .map(line => '"' + line.join(' ') + '"')
    .join(' ');

  readonly ROUTE_ROOT = ROUTE_DASHBOARD;

  readonly ROUTES: RouteDef[] = [
    {
      icon: 'overview',
      route: ROUTE_OVERVIEW,
      label: 'Overview',
      subs: [{icon: 'current', route: ROUTE_CURRENT, label: 'Current', subs: []}],
    },
    {
      icon: 'graph',
      route: ROUTE_GRAPH,
      label: 'Graph',
      subs: [{icon: 'walker', route: ROUTE_WALKER, label: 'Walker', subs: []}],
    },
    {
      icon: 'cryptocurrency',
      route: ROUTE_CRYPTO,
      label: 'Cryptocurrency',
      subs: [{icon: 'items1', route: ROUTE_BLOCKCHAIN, label: 'Blockchain', subs: []}],
    },
    {
      icon: 'game',
      route: ROUTE_GAME,
      label: 'Game',
      subs: [
        {icon: 'items1', route: ROUTE_GAME_MINESWEEPER, label: 'Minesweeper', subs: []},
        {icon: 'items2', route: ROUTE_GAME_SNAKE, label: 'Snake', subs: []},
        {icon: 'items3', route: ROUTE_GAME_DOWN, label: 'Down', subs: []},
        {icon: 'items4', route: ROUTE_RENDER_CANVAS, label: 'Render', subs: []},
      ],
    },
    {
      icon: 'playground',
      route: ROUTE_PLAYGROUND,
      label: 'Demo',
      subs: [
        {icon: 'items1', route: ROUTE_DEMO_MISC, label: 'Misc.', subs: []},
        {icon: 'items2', route: ROUTE_DEMO_STATE, label: 'State', subs: []},
        {icon: 'items3', route: ROUTE_DEMO_GHIBLI, label: 'Ghibli', subs: []},
      ],
    },
    {
      icon: 'settings',
      route: ROUTE_SETTINGS,
      label: 'Settings',
      subs: [
        {icon: 'configuration', route: ROUTE_CONFIGURATION, label: 'Configuration', subs: []},
        {icon: 'build_config', route: ROUTE_BUILDCONFIG, label: 'Build Settings', subs: []},
      ],
    },
  ];

  readonly isVisibleFooter$ = this.rxState.watch(state => state.ui.dashboard.isVisibleFooter, this.done$);
  readonly isVisibleHeader$ = this.rxState.watch(state => state.ui.dashboard.isVisibleHeader, this.done$);
  readonly isVisibleSide$ = this.rxState.watch(state => state.ui.dashboard.isVisibleSide, this.done$);

  readonly subRoute$ = this.rxState
    .watch(state => state.globalValues.route, this.done$)
    .pipe(
      map(url => {
        const paths = url.split('/');
        const indexMe = paths.indexOf(ROUTE_DASHBOARD);
        return indexMe >= 0 && paths.length > indexMe + 1 ? paths[indexMe + 1] : null;
      }),
    );

  readonly subRoutes$ = this.subRoute$.pipe(
    map(subRoute => {
      const route = this.ROUTES.find(ii => ii.route === subRoute);
      return route ? route.subs : [];
    }),
  );

  ngOnDestroy() {}
}
