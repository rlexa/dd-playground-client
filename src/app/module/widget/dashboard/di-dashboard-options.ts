import {InjectionToken} from '@angular/core';
import {BehaviorSubject} from 'rxjs';
import {StateSubject} from 'src/app/util/state-subject';

/** UI Dashboard visibility of header. */
export const DiDashboardVisibilityHeader = new InjectionToken<BehaviorSubject<boolean>>('DI Dashboard Visibility Header.', {
  providedIn: 'root',
  factory: () => new StateSubject(true),
});

/** UI Dashboard visibility of footer. */
export const DiDashboardVisibilityFooter = new InjectionToken<BehaviorSubject<boolean>>('DI Dashboard Visibility Footer.', {
  providedIn: 'root',
  factory: () => new StateSubject(true),
});

/** UI Dashboard visibility of sidebar. */
export const DiDashboardVisibilitySidebar = new InjectionToken<BehaviorSubject<boolean>>('DI Dashboard Visibility Sidebar.', {
  providedIn: 'root',
  factory: () => new StateSubject(true),
});
