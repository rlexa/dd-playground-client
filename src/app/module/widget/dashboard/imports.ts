import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
