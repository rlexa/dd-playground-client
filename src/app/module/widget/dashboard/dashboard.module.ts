import {CommonModule} from '@angular/common';
import {NgModule} from '@angular/core';
import {RouterModule, Routes} from '@angular/router';
import {ROUTE_ROOT, ROUTE_WILDCARD} from 'src/app/routing';
import {FooterModule} from '../footer';
import {NavigationBarItem, NavigationBarItemsData, NavigationBarModule} from '../navigation-bar';
import {DashboardRoute} from './dashboard-route';
import {DashboardComponent} from './dashboard.component';

const routeNavs: Record<DashboardRoute, NavigationBarItem> = {
  [DashboardRoute.Crypto]: {icon: 'cryptocurrency', route: DashboardRoute.Crypto, label: 'Cryptocurrency'},
  [DashboardRoute.Game]: {icon: 'game', route: DashboardRoute.Game, label: 'Game'},
  [DashboardRoute.Graph]: {icon: 'graph', route: DashboardRoute.Graph, label: 'Graph'},
  [DashboardRoute.Overview]: {icon: 'overview', route: DashboardRoute.Overview, label: 'Overview'},
  [DashboardRoute.Playground]: {icon: 'playground', route: DashboardRoute.Playground, label: 'Demo'},
  [DashboardRoute.School]: {icon: 'school', route: DashboardRoute.School, label: 'School'},
  [DashboardRoute.Settings]: {icon: 'settings', route: DashboardRoute.Settings, label: 'Settings'},
};

const data: NavigationBarItemsData = {
  navigationBarItems: Object.values(DashboardRoute).map((ii) => routeNavs[ii]),
};

const ROUTING: Routes = [
  {
    path: ROUTE_ROOT,
    component: DashboardComponent,
    data,
    children: [
      {
        path: DashboardRoute.Overview,
        loadChildren: () => import('src/app/module/widget/route-overview/route-overview.module').then((m) => m.RouteOverviewModule),
      },
      {
        path: DashboardRoute.Game,
        loadChildren: () => import('src/app/module/widget/route-game/route-game.module').then((m) => m.RouteGameModule),
      },
      {
        path: DashboardRoute.Graph,
        loadChildren: () => import('src/app/module/widget/graph-top-level/graph-top-level.module').then((m) => m.GraphTopLevelModule),
      },
      {
        path: DashboardRoute.Crypto,
        loadChildren: () => import('src/app/module/widget/route-crypto/route-crypto.module').then((m) => m.RouteCryptoModule),
      },
      {
        path: DashboardRoute.School,
        loadChildren: () => import('src/app/module/widget/route-school/route-school.module').then((m) => m.RouteSchoolModule),
      },
      {
        path: DashboardRoute.Playground,
        loadChildren: () => import('src/app/module/widget/route-playground/route-playground.module').then((m) => m.RoutePlaygroundModule),
      },
      {
        path: DashboardRoute.Settings,
        loadChildren: () => import('src/app/module/widget/route-system/route-system.module').then((m) => m.RouteSystemModule),
      },
      {path: ROUTE_ROOT, redirectTo: DashboardRoute.Overview, pathMatch: 'full'},
      {path: ROUTE_WILDCARD, redirectTo: DashboardRoute.Overview},
    ],
  },
];

@NgModule({
  imports: [CommonModule, FooterModule, NavigationBarModule, RouterModule.forChild(ROUTING)],
  exports: [DashboardComponent],
  declarations: [DashboardComponent],
})
class DashboardModule {}

export {DashboardModule, DashboardComponent};
