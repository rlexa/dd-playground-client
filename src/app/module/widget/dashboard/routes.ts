import {Route} from '@angular/router';
import {RouteRoot, RouteWild} from 'src/app/routing';
import {NavigationBarItem, NavigationBarItemsData} from '../navigation-bar';
import {DashboardRoute} from './dashboard-route';
import {DashboardComponent} from './dashboard.component';

const routeNavs: Record<DashboardRoute, Omit<NavigationBarItem, 'route'>> = {
  [DashboardRoute.Crypto]: {icon: 'cryptocurrency', label: 'Cryptocurrency'},
  [DashboardRoute.Game]: {icon: 'game', label: 'Game'},
  [DashboardRoute.Graph]: {icon: 'graph', label: 'Graph'},
  [DashboardRoute.Overview]: {icon: 'overview', label: 'Overview'},
  [DashboardRoute.Playground]: {icon: 'playground', label: 'Demo'},
  [DashboardRoute.School]: {icon: 'school', label: 'School'},
  [DashboardRoute.Settings]: {icon: 'settings', label: 'Settings'},
};

const data: NavigationBarItemsData = {
  navigationBarItems: Object.values(DashboardRoute).map<NavigationBarItem>((ii) => ({...routeNavs[ii], route: ii})),
};

export default [
  {
    path: RouteRoot,
    component: DashboardComponent,
    data,
    children: [
      {path: DashboardRoute.Overview, loadChildren: () => import('../../feature/overview/routes')},
      {path: DashboardRoute.Game, loadChildren: () => import('../../feature/game/routes')},
      {path: DashboardRoute.Graph, loadChildren: () => import('../../widget/graph-top-level/routes')},
      {path: DashboardRoute.Crypto, loadChildren: () => import('../../feature/crypto/routes')},
      {path: DashboardRoute.School, loadChildren: () => import('../../feature/school/routes')},
      {path: DashboardRoute.Playground, loadChildren: () => import('../../feature/playground/routes')},
      {path: DashboardRoute.Settings, loadChildren: () => import('../../feature/system/routes')},
      {path: RouteRoot, redirectTo: DashboardRoute.Overview, pathMatch: 'full'},
      {path: RouteWild, redirectTo: DashboardRoute.Overview},
    ],
  },
] as Route[];
