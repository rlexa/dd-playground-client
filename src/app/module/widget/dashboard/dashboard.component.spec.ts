import {async, TestBed} from '@angular/core/testing';
import {MatButton} from '@angular/material/button';
import {MatToolbar} from '@angular/material/toolbar';
import {MatTooltip} from '@angular/material/tooltip';
import {RouterTestingModule} from '@angular/router/testing';
import {MockComponents, MockDirectives, MockPipe} from 'ng-mocks';
import {provideRxState} from 'src/app/rx-state/test';
import {detectChanges, overrideForChangeDetection} from 'src/app/test';
import {FlexboxDirective} from '../../directive/flexbox';
import {IconPipe} from '../../pipe/icon';
import {VersionComponent} from '../version';
import {DashboardComponent} from './dashboard.component';

describe('DashboardComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [
        DashboardComponent,
        MockComponents(MatButton, MatToolbar),
        MockDirectives(MatTooltip, FlexboxDirective),
        MockComponents(VersionComponent),
        MockPipe(IconPipe, val => `icon ${val}`),
      ],
      providers: [provideRxState],
    })
      .overrideComponent(DashboardComponent, overrideForChangeDetection)
      .compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(DashboardComponent);
    detectChanges(fixture);
    expect(fixture).toMatchSnapshot();
  });
});
