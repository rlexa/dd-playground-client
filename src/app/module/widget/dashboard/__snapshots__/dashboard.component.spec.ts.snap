// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardComponent is created 1`] = `
<app-dashboard
  GRID={[Function String]}
  GRID_CONTENT={[Function String]}
  GRID_FOOTER={[Function String]}
  GRID_HEAD={[Function String]}
  GRID_SIDE={[Function String]}
  ROUTES={[Function Array]}
  ROUTE_ROOT={[Function String]}
  done$={[Function DoneSubject]}
  isVisibleFooter$={[Function BehaviorSubject]}
  isVisibleHeader$={[Function BehaviorSubject]}
  isVisibleSide$={[Function BehaviorSubject]}
  rxState={[Function RxStateService]}
  subRoute$={[Function AnonymousSubject]}
  subRoutes$={[Function AnonymousSubject]}
>
  <div
    class="mat-app-background position-absolute match-parent"
    style="display: grid; grid-template-areas: \\"head head\\" \\"side cont\\" \\"foot foot\\"; grid-template-columns: max-content auto; grid-template-rows: max-content auto max-content;"
  >
    <div
      style="grid-area: head; background: blue;"
    >
      
      
      <mat-toolbar
        class="mat-toolbar mat-primary mat-toolbar-single-row"
        color="primary"
      >
        <span
          class="mat-toolbar-filler"
        />
        <nav
          class="mat-nav-list"
        >
          
          
          <a
            aria-disabled="false"
            class="mat-button mat-button-base _mat-animation-noopable"
            href="/dashboard/overview"
            mat-button=""
            routerlinkactive="active"
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); min-width: auto;"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons transition-transform active-scale"
              >
                widgets
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
          
          
          <a
            aria-disabled="false"
            class="mat-button mat-button-base _mat-animation-noopable"
            href="/dashboard/graph"
            mat-button=""
            routerlinkactive="active"
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); min-width: auto;"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons transition-transform active-scale"
              >
                blur_on
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
          
          
          <a
            aria-disabled="false"
            class="mat-button mat-button-base _mat-animation-noopable"
            href="/dashboard/crypto"
            mat-button=""
            routerlinkactive="active"
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); min-width: auto;"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons transition-transform active-scale"
              >
                monetization_on
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
          
          
          <a
            aria-disabled="false"
            class="mat-button mat-button-base _mat-animation-noopable"
            href="/dashboard/game"
            mat-button=""
            routerlinkactive="active"
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); min-width: auto;"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons transition-transform active-scale"
              >
                games
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
          
          
          <a
            aria-disabled="false"
            class="mat-button mat-button-base _mat-animation-noopable"
            href="/dashboard/playground"
            mat-button=""
            routerlinkactive="active"
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); min-width: auto;"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons transition-transform active-scale"
              >
                videogame_asset
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
          
          
          <a
            aria-disabled="false"
            class="mat-button mat-button-base _mat-animation-noopable"
            href="/dashboard/settings"
            mat-button=""
            routerlinkactive="active"
            style="user-select: none; webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0); min-width: auto;"
            tabindex="0"
          >
            <span
              class="mat-button-wrapper"
            >
              <i
                class="material-icons transition-transform active-scale"
              >
                settings
              </i>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </a>
          
        </nav>
        <span
          class="mat-toolbar-filler"
        />
      </mat-toolbar>
    </div>
    <div
      class="position-relative"
      style="grid-area: side;"
    >
      
      
      <mat-toolbar
        class="toolbar-column mat-toolbar mat-primary mat-toolbar-single-row"
        color="primary"
      >
        
        <span
          class="mat-toolbar-filler"
        />
      </mat-toolbar>
    </div>
    <main
      class="position-relative"
      style="grid-area: cont;"
    >
      <router-outlet />
    </main>
    <footer
      style="grid-area: foot;"
    >
      
      <div
        class="border-top contenthv"
        ddflexbox="hycr"
        style="display: flex; flex-direction: row-reverse; justify-content: flex-start; align-items: center; align-content: stretch; flex-wrap: no-wrap;"
      >
        <app-version
          class="margin-bottom-narrow"
        >
          <button
            class="auto-line-height mat-button mat-button-base _mat-animation-noopable"
            mat-button=""
          >
            <span
              class="mat-button-wrapper"
            >
              <label
                class="deadclick-force"
              >
                v null
              </label>
            </span>
            <div
              class="mat-button-ripple mat-ripple"
              matripple=""
            />
            <div
              class="mat-button-focus-overlay"
            />
          </button>
        </app-version>
      </div>
    </footer>
  </div>
</app-dashboard>
`;
